src/reducers/images.js

```diff
@@ -64,6 +70,28 @@ const images = (state = initialState, { type, payload }) => {
         }
       };
 
+    case UPDATE_TEXT:
+      return {
+        ...state,
+        ...{
+          caption: payload.text
+        }
+      };
+
+    case UPDATE_TEXT_COLOR:
+      return {
+        ...state,
+        fontColor: payload.fontColor
```

Comment:
open question as to whether we want this pattern or whether we should spread out a new object with `fontColor`